clar2wasm/src/words/sequences.rs

```diff
@@ -565,8 +565,22 @@ impl ComplexWord for Map {
                 .clone()
             {
                 TypeSignature::SequenceType(SequenceSubtype::ListType(lt)) => {
-                    let element_ty = lt.get_list_item_type().clone();
-                    let element_size = get_type_size(&element_ty);
+                    let item_ty = &lt.get_list_item_type().clone();
+                    let element_size = get_type_size(item_ty);
+
+                    // Fix for an issue when dealing with responses and optional types.
+                    // Set the type of the list to the type of the function argument.
+                    let element_ty = match item_ty {
+                        TypeSignature::ResponseType(_) => TypeSignature::ResponseType(Box::new((
+                            return_element_type.clone(),
+                            return_element_type.clone(),
```

Comment:
I'm not really familiar with the inner workings, but this seems strange: IIUC this is to cover the `NoType` case, but why are you typing an input element with the output type? Also, isn't this always overwriting types, even if they are known?

Reply:
Changed the implementation. I really don't know when my head was while working on that. :-p